### ISSUE TITLE
EPA CEMS Intake Catalog

### ISSUE THREAD

**Issue by zaneselvans** [catalyst-cooperative/pudl#1564]
# Description
Create a full featured Intake Catalog for distributing the EPA CEMS hourly emissions data stored as Parquet files. This follows some exploration in #1155. See also notes in #1495 and PR #1563

## Billing
This work should be billed under our Sloan Foundation "Data Distribution" sub-project.

## Goals
* Allow anonymous / public users to easily work with this relatively large data set (~1 billion rows, 50GB uncompressed).
* Make it fast and easy to query subsets of the dataset remotely, without having to download the whole thing.
* Provide human and machine readable metadata in association with the data itself, so that users can understand what they're working with.
* Ensure that parquet file names are human & machine readable, and compatible with many different filesystems and parquet access methods.
* Ensure that our EPA CEMS ETL outputs data that's appropriate for distribution as a data catalog.
* Avoid unnecessary data storage and egress fees.

### Phase 1:
Get a functional intake catalog deployed for demonstration & feedback.
* [x] Switch to producing a single EPA CEMS Parquet file as output, with (year, state) row groups (See #1563).
* [x] Ensure that all column types are preserved by Intake as they would be by `read_parquet()`
* [x] Create a dedicated repository for our high-level data catalog.
* [x] Enable local caching of remote data using `fsspec` / `simplecache` (see comment on #1496)
* [x] Enable output of year-state partitioned files in the EPA CEMS ETL.
* [x] Package `pudl_catalog` for installation using `pip`.
* [x] https://github.com/catalyst-cooperative/pudl-catalog/issues/5
* [x] https://github.com/catalyst-cooperative/pudl-catalog/issues/6
* [x] https://github.com/catalyst-cooperative/pudl-catalog/issues/1
* [x] https://github.com/catalyst-cooperative/pudl-catalog/issues/11
* [x] https://github.com/catalyst-cooperative/pudl-catalog/issues/10

### Phase 2:
Flesh out metadata and improve performance.
* [x] https://github.com/catalyst-cooperative/pudl-catalog/issues/8 (Consolidated with [issue 4](https://github.com/catalyst-cooperative/pudl-catalog/issues/4) below)
* [ ] https://github.com/catalyst-cooperative/pudl-catalog/issues/4
* [ ] https://github.com/catalyst-cooperative/pudl-catalog/issues/7
* [ ] https://github.com/catalyst-cooperative/pudl-catalog/issues/2

## Out of Scope
* Distributed processing of the EPA CEMS data (that'll happen in the Dagster refactor).
* Additional EPA CEMS data wrangling, e.g. the things discussed in #1049 
* https://github.com/catalyst-cooperative/pudl-catalog/issues/3

**Comment by zaneselvans** [comment]
Whoops yes I forgot to add the intake requirements. I had them installed in my local environment.

The dtypes you've got listed there seem to be the correct ones. But you have no `year` or `state` columns.  I'm still confused as to why those aren't showing up, given that the data is definitely stored in the files.

I really don't understand how the source specific metadata works. My suspicion is that the allowable year/state values can be put in there, and the column/table descriptions, but I don't see any documentation on how to do it appropriately.

**Comment by zaneselvans** [comment]
Hey @martindurant thanks so much for your comment on #1496! I got `simplecache` working and created a basic installable catalog, and have been experimenting with different setups for our open US energy data catalog over in [the pudl-catalog repo](https://github.com/catalyst-cooperative/pudl-catalog). I've collected a bunch of outstanding issues in the issue above, which point at the individual `pudl-catalog` issues and was wondering if we might be able to get some advice from you on how best to set things up. I'm not sure which of these things is just me not understanding how to configure the catalog correctly and which are deeper constraints.

Do you happen to have a list of publicly visible intake catalogs that use Parquet data sources? I've tried searching GitHub but haven't been very successful. The [CarbonPlan Data repo](https://github.com/carbonplan/data) is the best I've seen, but they have a very simple configuration.

Once the EPA CEMS Hourly Emissions data source is finished, we also want to look at writing an intake-sqlite driver (#1156) to manage the distribution of versioned SQLite databases, which will download and cache the database file locally, and then use the `intake-sql` driver to access it. Does that seem like a reasonable approach?  Thanks for any pointers you can offer!

**Comment by zaneselvans** [comment]
Closing this since we ultimately decided against using intake catalogs.
